internal/crypto/src/openssl/check_certificate_trust.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License,
+// Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+// or the MIT license (http://opensource.org/licenses/MIT),
+// at your option.
+
+// Unless required by applicable law or agreed to in writing,
+// this software is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR REPRESENTATIONS OF ANY KIND, either express or
+// implied. See the LICENSE-MIT and LICENSE-APACHE files for the
+// specific language governing permissions and limitations under
+// each license.
+
+use openssl::{
+    stack::Stack,
+    x509::{verify::X509VerifyFlags, X509StoreContext, X509},
+};
+
+use crate::{
+    cose::{CertificateTrustError, CertificateTrustPolicy},
+    openssl::OpenSslMutex,
+};
+
+pub(crate) fn check_certificate_trust(
+    ctp: &CertificateTrustPolicy,
+    chain_der: &[Vec<u8>],
+    cert_der: &[u8],
+    signing_time_epoch: Option<i64>,
+) -> Result<(), CertificateTrustError> {
+    // First check to see if the certificate appears on the allowed list of
+    // end-entity certificates.
+    if ctp.end_entity_cert_ders().any(|der| der == cert_der) {
```

Comment:
Looks like you removed this behavior:

// check the cert against the allowed list first
    let cert_sha256 = hash_sha256(ee_der);
    let cert_hash_base64 = base64::encode(&cert_sha256);
    if th.get_allowed_list().contains(&cert_hash_base64) {
        return Ok(true);
    }
This was an optimization we use for the Verify site.  Verify only stores the hashes of the der encoded pems.  That greatly reduced the size of the payload and speed of lookup that Verify performs.

Reply:
Same thing in the wasm case.

Reply:
Ah, I wasn't aware of that use case, but I'll add it back.

Reply:
If you look at load_allowed_list in the old versions you will see we allow you to use either PEM or DER hashes.  Verify uses DER hashes.

Reply:
Note this code should go away one day when we stop supporting end-entity certs.

Reply:
Actually, if I read this right, we were already using DERs under the hood here. Which means Verify should be fine. And makes sense since the underlying API only accepts DER-encoded end-entity certs.

So I think this is OK as is. (Happy to change after merging if that's wrong.)